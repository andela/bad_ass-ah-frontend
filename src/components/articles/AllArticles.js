import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import getAllArticle from '../../actions/article';
import Layout from '../layouts/Layout';

export class AllArticles extends Component {
  componentDidMount() {
    const { getAllArticle } = this.props;
    getAllArticle();
  }

  render() {
    const { article } = this.props;
    const { articles } = article;
    return (
      <Layout>
        <Fragment>
          {articles !== undefined ? (
            <Fragment>
              {articles.length > 0 ? articles.map(article => <div key={article.id}>{article.title}</div>) : 'Not found'}
            </Fragment>
          ) : 'No article found'}
        </Fragment>
      </Layout>
    );
  }
}

AllArticles.propTypes = {
  getAllArticle: PropTypes.func.isRequired,
  article: PropTypes.func.isRequired
};
// mapStateToProps from reducer
const mapStateToProps = state => ({
  article: state.articles
});

export default connect(mapStateToProps, { getAllArticle })(AllArticles);
