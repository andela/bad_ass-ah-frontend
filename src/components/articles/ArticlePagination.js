
import React, { Component } from 'react';

export default class ArticlePagination extends Component {
  // eslint-disable-next-line class-methods-use-this
  render() {
    return (
            <div className="pagination">
                <span>&laquo;</span>
                <span className="active-page">1</span>
                <span>1</span>
                <span>2</span>
                <span>3</span>
                <span>4</span>
            </div>
    );
  }
}
