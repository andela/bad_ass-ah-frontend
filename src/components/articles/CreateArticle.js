import React, { Component } from 'react';
import PropType from 'prop-types';
import { connect } from 'react-redux';
import ReactFileReader from 'react-file-reader';
import { createArticle, addTag, removeTag } from '../../actions/article';
import Layout from '../layouts/Layout';
import Loading from '../layouts/Loading';

export class CreateArticle extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      image: '',
      tag: '',
      loading: false,
      open: false
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleFiles = this.handleFiles.bind(this);
    this.addTag = this.addTag.bind(this);
    this.openTag = this.openTag.bind(this);
    this.destroyTag = this.destroyTag.bind(this);
    this.BodyContent = React.createRef();
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    // @set loading to true
    this.setState({ loading: true });
    const { title, fileImg } = this.state;
    const { create } = this.props;
    const data = {
      title,
      body: this.BodyContent.current ? this.BodyContent.current.innerHTML : '',
      image: fileImg,
      tag: create.newTag
    };
    const { createArticle } = this.props;
    createArticle(data);
  }

setBold=() => {
  document.execCommand('Bold', false, null);
}

openTag(e) {
  e.preventDefault();
  const { open } = this.state;
  this.setState({ open: !open });
}

addTag(e) {
  e.preventDefault();
  const { addTag } = this.props;
  const { tag } = this.state;
  addTag(tag);
  this.setState({ tag: '' });
}

handleFiles(files) {
  this.setState({ image: files || '', fileImg: files.fileList[0] });
}

destroyTag(tag) {
  const { removeTag } = this.props;
  removeTag(tag);
}

render() {
  let successMessage;
  const {
    title, image, loading, tag, open
  } = this.state;
  const { create } = this.props;
  if (Object.keys(create).length !== 0 && create.newArticle !== null) {
    if (create.newArticle !== undefined) successMessage = create.newArticle.message;
    else successMessage = '';
    setTimeout(() => {
      window.location.reload(true);
    }, 2000);
  }
  return (
    <Layout>
      <Loading loading={loading} message={successMessage} />
      <div className="G-showcase">
        <div className="G-create-article" data-test="G-create-article">
          <div className="G-actionType">
            {open === true && (
            <div className="G-tags">
              <input type="text" name="tag" placeholder="Add tag" onChange={this.onChange} value={tag} />
              <button type="button" className="G-tagBtn" onClick={this.addTag}>Add</button>
            </div>
            )}
            <div className="G-action">
              <button
                className="setBold"
                type="button"
                onClick={this.setBold}
                data-test="G-bold"
              >
                <i className="icofont-bold" />
              </button>
            </div>
            <div className="G-action">
              <button
                type="button"
                className="setItalic"
                onClick={() => document.execCommand('Italic', false, null)}
              >
                <i className="icofont-italic" />
              </button>
            </div>
            <div className="G-action">
              <ReactFileReader handleFiles={this.handleFiles} base64 data-test="G-image">
                <button type="button" className="setImage">
                  <i className="icofont-ui-image uploadImageIcon" />
                </button>
              </ReactFileReader>
            </div>
            <div className="G-action">
              <button type="button" onClick={this.openTag}>
                <i className="icofont-plus" />
                <small>Tags</small>
              </button>
            </div>
            <div className="G-publish">
              <button type="button" onClick={this.onSubmit} data-test="G-submitData">
                <i className="icofont-world worldIcon" />
                {' '}
                Publish
              </button>
            </div>
          </div>
          <div className="G-form-group">
            <input
              type="text"
              name="title"
              id=""
              className="G-input"
              placeholder="title"
              value={title}
              onChange={this.onChange}
              data-test="G-input"
            />
          </div>
          {create.newTag !== undefined && create.newTag.length !== 0 && (
          <div className="G-form-group">
            {create.newTag.map(tag => (
              <div className="G-displayTag" key={Math.random()}>
                <div>{tag}</div>
                <div
                  className="G-removeTag"
                  data-test="G-destroyTag"
                  role="button"
                >
                  <button type="button" onClick={this.destroyTag.bind(this, tag)} className="trashBtn">
                    <i className="icofont-trash" />
                  </button>
                </div>
              </div>
            ))}
          </div>
          )}
          <div className="G-form-group">
            {image && (
              <div className="G-image-display">
                <img src={image.base64} alt="selected " />
                {' '}
              </div>
            )}
          </div>
          <div className="G-form-group">
            <div
              id="texteditor"
              name="body"
              ref={this.BodyContent}
              contentEditable="true"
            />

          </div>
        </div>
      </div>
    </Layout>
  );
}
}
CreateArticle.propTypes = {
  createArticle: PropType.func.isRequired,
  addTag: PropType.func.isRequired,
  removeTag: PropType.func.isRequired,
  create: PropType.objectOf(PropType.object).isRequired
};
const mapStateToProps = state => ({
  create: state.articles
});
export default connect(mapStateToProps, { createArticle, addTag, removeTag })(CreateArticle);
