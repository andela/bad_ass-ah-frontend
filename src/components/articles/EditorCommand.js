import React, { Component, Fragment } from 'react';

class Helper extends Component {
     state={
       link: '',
       open: false,
       display: false,
     }

     onChange= (e) => {
       this.setState({ [e.target.name]: e.target.value });
     }

     setBold=() => {
       document.execCommand('Bold', false, null);
     }

     openTag=(e) => {
       e.preventDefault();
       this.setState(prevState => ({
         open: !prevState.open
       }));
     }

     addLink= (link) => {
       document.execCommand('CreateLink', false, link);
     }

     render() {
       const { open, link } = this.state;
       return (
            <Fragment>
              {open && (
              <div className="G-links">
                <input type="text" name="link" placeholder="place your link here"
                data-test="G-onChange"
                onChange={this.onChange} value={link} />
                <button
                  type="button"
                  className="G-tagBtn"
                  data-test="Add-link"
                  onClick={this.addLink.bind(this, link)}
                >
                Add
                </button>
              </div>
              )}
              <div className="G-action">
                <button
                  className="setBold"
                  type="button"
                  onClick={this.setBold}
                  data-test="G-bold"
                >
                  <i className="icofont-bold" />
                </button>
              </div>
              <div className="G-action">
                <button
                  type="button"
                  className="setItalic"
                  onClick={() => document.execCommand('Italic', false, null)}
                >
                  <i className="icofont-italic" />
                </button>
              </div>
              <div className="G-action">
                <button type="button" data-test="G-openTag" onClick={this.openTag}>
                  <i className="icofont-link" />
                  <small>link</small>
                </button>
              </div>
            </Fragment>
       );
     }
}

export default Helper;
