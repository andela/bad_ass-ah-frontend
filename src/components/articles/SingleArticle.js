/* eslint-disable no-unused-vars */
/* eslint-disable array-callback-return */
/* eslint-disable class-methods-use-this */
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import htmlParser from 'html-react-parser';
import stringParser from 'react-to-string';
import Hashids from 'hashids';
import { Link, Redirect } from 'react-router-dom';
import { likeArticle, dislikeArticle } from '../../actions/voteArticle';
import { singleArticle, deleteArticle } from '../../actions/article';
import Layout from '../layouts/Layout';
import NotFound from '../NotFound';
import Comment from '../comment/comments';
import Rating from './rating/Rating';
import { bookmarkArticle } from '../../actions/bookmarkArticle';
import { isAuthenticated } from '../../helpers/Config';
import { getComments } from '../../actions/comment/comment';
import { setReadingStats } from '../../actions';
import ShareArticle from './shareArticle/shareArticle';
import HighlightPopover from '../popovers/Highlight';
import CommentPopover from '../popovers/Comment';
import Alert from '../layouts/Alert';
import { highlightText, getUserHighlights } from '../../actions/highlight';
import { markHighlightedText } from '../../utils/markHighlightedText';
import { setLoginRedirectPath } from '../../actions/login';
import Spinner from '../layouts/Spinner';
import Report from './reporting/Report';

const hashids = new Hashids('', 10);

export class SingleArticle extends Component {
  state = {
    articleId: '',
    hasLikedClass: null,
    hasDilikedClass: null,
    hasbookmarkedClass: null,
    bookmarkArticle: null,
    articleId2: null,
    userId: '',
    prevPath: '',
    shareArticleUrl: '',
    indexStart: '',
    indexEnd: '',
    text: '',
    comment: '',
    handle: this.props.match.params.handle,
    redirectOnBookmark: false
  };

  async componentWillMount() {
    // eslint-disable-next-line react/prop-types
    const {
      match: { params, url },
      getComments
    } = this.props;
    this.setState({ articleId2: params.handle });
    this.getArticle();
    const user = await isAuthenticated();
    this.setState({ userId: user.payload.id, shareArticleUrl: url }, () => {
      getComments(this.state.articleId2);
    });
  }

  getArticle = () => {
    const {
      match: { params },
      singleArticle,
      getUserHighlights
    } = this.props;
    this.setState({ articleId: params.handle });
    singleArticle(params.handle);
    getUserHighlights(params.handle);
  };

  async componentDidMount() {
    await this.getArticle();
    this.setReadingStats();
    // this.props.getComments(this.state.articleId2);
  }

  componentDidUpdate() {
    const elements = document.getElementsByClassName('highlighted');
    const topScroll = window.pageYOffset || document.documentElement.scrollTop;

    if (elements) {
      Array.from(elements).forEach((element) => {
        const rect = element.getBoundingClientRect();
        // eslint-disable-next-line no-param-reassign
        element.onclick = () => {
          const { highlights } = this.props;
          highlights.forEach((highlight) => {
            if (element.textContent === highlight.text) {
              const root = document.getElementsByClassName('App')[0];
              const div = document.createElement('div');
              div.className = 'highlight-comment';
              div.style.top = `${rect.top + topScroll}px`;
              div.style.left = `${rect.right}px`;
              div.innerHTML = ` <div class="title">
                                <div class="title-text">Comment</div>
                              </div>
                              <div class="comment">${highlight.comment}</div>`;
              root.appendChild(div);
              root.onmouseup = () => {
                div.classList.add('hide');
              };
            }
          });
        };
      });
    }
  }

  componentWillReceiveProps(nextProps) {
    const { voteMessage } = nextProps.articles;
    // eslint-disable-next-line react/prop-types
    const { location } = nextProps;
    if (location !== undefined) {
      const path = location.state ? location.state.prevPath : '/';
      this.setState({ prevPath: path });
    }
    if (voteMessage === 'thanks for the support.') {
      this.setState({ hasLikedClass: 'changeColor', hasDilikedClass: null });
    }
    if (voteMessage === 'You have disliked this article.') {
      this.setState({ hasLikedClass: null, hasDilikedClass: 'changeColor' });
    }
  }

  setReadingStats = () => {
    if (this.props.isAuth) {
      this.props.setReadingStats(this.state.articleId2);
    }
  };

  likeArticle = async () => {
    const { articleId } = this.state;
    const { likeArticle } = this.props;
    await likeArticle(articleId);
    this.getArticle();
  };

  dislikeArticle = async () => {
    const { articleId } = this.state;
    const { dislikeArticle } = this.props;
    await dislikeArticle(articleId);
    this.getArticle();
  };

  bookmarks = async () => {
    if (this.props.isAuth) {
      const { articleId } = this.state;
      const { bookmarkArticle } = this.props;
      await bookmarkArticle(articleId);
      this.getArticle();
    } else {
      this.props.setLoginRedirectPath(window.location.pathname);
      this.setState({ redirectOnBookmark: true });
    }
  }

  onSelectedText = () => {
    const popup = document.getElementById('highlight-popup');
    document.onmouseup = () => {
      const selection = document.getSelection();
      const selectedText = selection.toString();

      if (selection.rangeCount > 0) {
        const range = selection.getRangeAt(0);
        const rect = range.getBoundingClientRect();
        const topScroll = window.pageYOffset || document.documentElement.scrollTop;

        if (selectedText.length) {
          popup.classList.remove('hide');
          popup.style.top = `${rect.top + topScroll - 80}px`;
          popup.style.left = `${rect.left}px`;

          this.setState({
            indexStart: selection.anchorOffset,
            indexEnd:
              selection.anchorOffset === 0
                ? selection.anchorOffset + selectedText.length - 1
                : selection.anchorOffset + selectedText.length,
            text: selectedText
          });
        } else {
          popup.classList.add('hide');
        }
      }
    };
  };

  onHighlight = () => {
    const popup = document.getElementById('highlight-popup');
    const comment = document.getElementById('comment-popover');
    const div = document.getElementsByClassName('G-showcase')[0];
    popup.classList.add('hide');
    const selection = window.getSelection();
    const range = selection.getRangeAt(0);
    const rect = range.getBoundingClientRect();
    const topScroll = window.pageYOffset || document.documentElement.scrollTop;
    const newNode = document.createElement('span');
    newNode.style.background = 'aqua';
    range.surroundContents(newNode);
    comment.classList.remove('hide');
    comment.style.top = `${rect.top + topScroll}px`;
    comment.style.left = `${rect.right - 40}px`;

    div.onmouseup = () => {
      comment.classList.add('hide');
      newNode.style.background = '';
    };
  };

  onChange = (e) => {
    this.setState({ comment: e.target.value });
  };

  onSubmit = (e) => {
    e.preventDefault();
    const { highlightText } = this.props;

    highlightText(this.state);
    window.getSelection().removeAllRanges();
    this.setState({
      indexStart: '',
      indexEnd: '',
      text: '',
      comment: ''
    });
  };

  destroy = (id) => {
    const { deleteArticle } = this.props;
    this.setState({ startLoading: true });
    deleteArticle(id);
  };

  render() {
    let single;
    const {
      prevPath, userId, articleId2, shareArticleUrl, redirectOnBookmark
    } = this.state;
    const {
      articles: { article, error, message },
      highlights
    } = this.props;
    if (article !== null) if (article && article.article !== undefined) single = article.article;

    this.onSelectedText();

    let redirectOnBookmarkURL = null;
    if (redirectOnBookmark) {
      redirectOnBookmarkURL = <Redirect to='/auth' />;
    }

    return (
      <Layout>
        {redirectOnBookmarkURL}
        {message !== '' && window.location.replace(prevPath)}
        <HighlightPopover onHighlight={this.onHighlight} />
        <CommentPopover onChange={this.onChange} onSubmit={this.onSubmit} />
        <div className="G-showcase">
          <Fragment>
            {single !== undefined ? (
              <div className="G-create-article" data-test="G-create-article">
                <Alert />
                <div className="G-form-group">
                  <h1 className="G-storyTitle">{stringParser(htmlParser(single.title))}</h1>
                  <p className='reading-time'>reading time: {single.readingTime}</p>
                  {userId === single.authorfkey.id && (
                    <div className="drop-article singleDrop">
                      <Link to={`/story/edit/${hashids.encode(single.article_id)}`}>
                        <button type="button" data-test="Btn-remove">
                          {' '}
                          <span>Edit</span>
                          <i className="icofont-ui-edit editIcon" />
                        </button>
                      </Link>
                      <button
                        type="button"
                        data-test="G-deleteArticle"
                        onClick={this.destroy.bind(this, hashids.encode(single.article_id))}
                      >
                        <span>Delete</span>
                        <i className="icofont-ui-delete deleteIcon" />
                      </button>
                    </div>
                  )}
                </div>
                {single.image && (
                  <div className="G-form-group">
                    <div className="G-image-display">
                      <img src={single.image} alt="selected " />{' '}
                    </div>
                  </div>
                )}
                {single.taglist !== null && single.taglist.length !== 0 ? (
                  <div className="G-form-group">
                    {single.taglist
                      && single.taglist.map(tag => (
                        <div className="G-displayTag" key={Math.random()}>
                          <div className="G-tagName"> {tag} </div>
                        </div>
                      ))}
                  </div>
                ) : (
                  ''
                )}
                <div className="G-form-group">
                  <div id="texteditor" name="body" className="G-singleEditor">
                    {highlights === null || highlights === undefined ? (
                      <Spinner />
                    ) : (
                      htmlParser(markHighlightedText(single.body, highlights))
                    )}
                  </div>
                </div>
                <div className="section__rating">
                  <Rating articleId={this.state.articleId2} />
                  <div className="C-like c-like-grid section__rating-like">
                    <div
                      id="like-btn"
                      className={`btn-heart ${this.state.hasLikedClass}`}
                      title="likes"
                      onClick={this.likeArticle}
                    >
                      {article.votes.hasLiked === true ? (
                        <i className="icofont-heart changeColor" />
                      ) : (
                          <i className="icofont-heart" />
                      )}
                      <div>{article.votes.likes}</div>
                    </div>
                    <div
                      id="dislike-btn"
                      className={`btn-heart ${this.state.hasDilikedClass}`}
                      title="dislikes"
                      onClick={this.dislikeArticle}
                    >
                      {article.votes.hasDisliked === true ? (
                        <i className="icofont-ui-love-broken changeColor" />
                      ) : (
                          <i className="icofont-ui-love-broken" />
                      )}
                      <div>{article.votes.dislikes}</div>
                    </div>
                    <div id="bookmark-btn" className={`btn-bookmark ${this.state.hasbookmarkedClass}`} title="bookmark" onClick={this.bookmarks}>
                      {article.hasBookmarked === true ? <i class="icofont-book-mark changeColor"></i> : <i class="icofont-book-mark"></i>}
                    </div>
                  </div>
                  <Report articleId={this.state.articleId2} />
                </div>
                <ShareArticle shareArticleUrl={shareArticleUrl} />
                <Comment articleId={articleId2} />
              </div>
            ) : (
                <center>
                  {error && error.errors !== undefined && <NotFound error={error.errors.body[0]} />}
                </center>
            )}
          </Fragment>
        </div>
      </Layout>
    );
  }
}
export const mapStateToProps = state => ({
  articles: state.articles,
  isAuth: state.login.isAuthenticated,
  highlights: state.highlight.highlights
});
SingleArticle.propTypes = {
  singleArticle: PropTypes.func.isRequired,
  articles: PropTypes.objectOf(PropTypes.object).isRequired,
  match: PropTypes.objectOf(PropTypes.object).isRequired,
  likeArticle: PropTypes.func.isRequired,
  dislikeArticle: PropTypes.func.isRequired,
  deleteArticle: PropTypes.func.isRequired,
  isAuth: PropTypes.bool.isRequired,
  getComments: PropTypes.func,
  setReadingStats: PropTypes.func.isRequired,
  highlightText: PropTypes.func,
  getUserHighlights: PropTypes.func,
  highlights: PropTypes.array,
  bookmarkArticle: PropTypes.func.isRequired,
  setLoginRedirectPath: PropTypes.func
};
// eslint-disable-next-line max-len
export default connect(
  mapStateToProps,
  {
    singleArticle,
    likeArticle,
    dislikeArticle,
    deleteArticle,
    getComments,
    setReadingStats,
    highlightText,
    getUserHighlights,
    bookmarkArticle,
    setLoginRedirectPath
  }
)(SingleArticle);
