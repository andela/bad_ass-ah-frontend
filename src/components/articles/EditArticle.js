
import React, { Component } from 'react';
import PropType from 'prop-types';
import ReactFileReader from 'react-file-reader';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import htmlParser from 'html-react-parser';
import {
  singleArticle, updateArticle
} from '../../actions/article';
import Layout from '../layouts/Layout';
import Loading from '../layouts/Loading';
import ArticleValidation from '../../helpers/Article';
import EditorCommand from './EditorCommand';

export class EditArticle extends Component {
 state = {
   image: '',
   tag: '',
   loading: false,
   titleError: '',
   bodyError: '',
   display: false,
   fileImg: '',
   handle: ''
 };

BodyContent = React.createRef();

Title = React.createRef();

componentDidMount() {
  const { match: { params }, singleArticle } = this.props;
  const { handle } = params;
  this.setState({ handle });
  singleArticle(handle);
}

  onSubmit=(e) => {
    e.preventDefault();
    this.setState({ loading: true, display: false });
    const { fileImg, handle } = this.state;
    const { articleReducer } = this.props;
    const data = {
      title: this.Title.current ? this.Title.current.innerHTML : '',
      body: this.BodyContent.current ? this.BodyContent.current.innerHTML : '',
      image: fileImg,
      tag: articleReducer.newTag
    };
    const validation = ArticleValidation(data);
    if (!validation) {
      const { updateArticle } = this.props;
      updateArticle(handle, data);
    } else {
      this.setState({
        titleError: validation.title,
        bodyError: validation.body,
        loading: false,
        display: true
      });
    }
  }

  handleFiles=(files) => {
    const data = files ? files.fileList[0] : '';
    this.setState({ image: files || '', fileImg: data });
  }

  render() {
    let successMessage;
    let single;
    const {
      image, loading, titleError, bodyError, display
    } = this.state;
    const { articleReducer } = this.props;
    const { message, article } = articleReducer;
    if (message !== '') {
      successMessage = 'Article updated successfully.';
    }
    if (article !== null && article !== undefined) {
      single = article.article;
    }
    return (
    <Layout>
      <Loading loading={loading} message={successMessage} />
      { successMessage !== undefined && <Redirect to="/view-profile"/>}
      <div className="G-showcase">
     <div className="G-create-article" data-test="G-create-article">
      <div className="G-containers">
      <div className="G-actionType">
          <div className="G-action">
                <ReactFileReader data-test="G-image" handleFiles={this.handleFiles} base64>
                  <button type="button" className="setImage">
                    <i className="icofont-ui-image uploadImageIcon" />
                  </button>
                </ReactFileReader>
              </div>
            <EditorCommand />
        </div>
        <div className="G-form-group">
              {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
              <label className="titleLabel changeTitle">Your title</label>
              <div contentEditable="true" ref={this.Title} data-test="G-input"
              name="title"
              className="G-titleEditor"
              suppressContentEditableWarning={true}>
               {single ? htmlParser(single.title) : ''}
              </div>
            </div>
            <div className="G-form-group">
              {single && single.image && (
              <div className="G-image-display">
                {image !== '' ? <img src={image.base64} alt="selected " /> : <img src={single.image ? single.image : ''} alt="selected " />}
                {' '}
              </div>
              )}
            </div>
            <div className="G-form-group">
              {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
              <label>Your content</label>
              <div
                id="texteditor"
                name="body"
                ref={this.BodyContent}
                contentEditable="true"
                suppressContentEditableWarning={true}
              >{single ? htmlParser(single.body) : ''}</div>
            </div>
            <div className="G-form-group">
              <div className="G-publish">
                <button type="button" onClick={this.onSubmit} data-test="G-updateData">
                  <i className="icofont-brand-appstore" />
                  {' '}
                save changes
                </button>
              </div>
            </div>
      </div>
      {display === true && (
          <div className="G-displayValidation">
            <div className="G-displayError" style={{ transform: 'translatex(0%)!important' }}>
              {titleError && (
              <div className="G-errors">
                <i className="icofont-lens" />
                {' '}
                {titleError}
              </div>
              )}
              {bodyError && (
              <div className="G-errors">
                <i className="icofont-lens" />
                {' '}
                {bodyError}
              </div>
              )}
            </div>
          </div>
      )}
      </div>
      </div>
    </Layout>
    );
  }
}
EditArticle.propTypes = {
  singleArticle: PropType.func.isRequired,
  articleReducer: PropType.objectOf(PropType.object).isRequired,
  updateArticle: PropType.func.isRequired,
  match: PropType.objectOf(PropType.object).isRequired,
  params: PropType.objectOf(PropType.object).isRequired
};
const mapStateToProps = state => ({
  articleReducer: state.articles
});
export default connect(mapStateToProps, {
  singleArticle, updateArticle
})(EditArticle);
