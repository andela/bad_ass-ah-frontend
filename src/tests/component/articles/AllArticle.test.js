import React from 'react';
import { shallow } from 'enzyme';
import { AllArticles } from '../../../components/articles/AllArticles';

//
const props = {
  getAllArticle: jest.fn(),
  article: jest.fn()
};

describe('Create Article', () => {
  const component = shallow(<AllArticles {...props} />);
  test('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });
});
