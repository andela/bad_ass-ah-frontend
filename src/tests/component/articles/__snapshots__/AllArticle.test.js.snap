// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Article should render without crashing 1`] = `ShallowWrapper {}`;
